DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -618,6 +617,10 @@ sealed class DMVisitorExpression : DMASTVisitor {
                                 operation.Identifier = field;
                                 operation.GlobalId = null;
                                 operation.Path = property.Type;
+                                if (operation.Kind == DMASTDereference.OperationKind.Field &&
```

Comment:
Checking `Kind` again here is kind of yucky but I'm not sure how else to do it without duplicating a bunch of this code in the other switch case.